### PR TITLE
FISH-1076 hotDeploy runtime option support in Payara Micro

### PR DESCRIPTION
**Pull request by jGauravGupta**
## Description
This is a feature PR to pass `hotDeploy` runtime option on Payara Micro startup to save deployment state and read `.reload` file to fetch modified files and metadata details.

### Testing Performed
Manually Tested with VSCode hotDeploy integration feature

### Testing Environment
VSCode 1.53.2, Windows 10, JDK 8

### CODE REVIEW

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/DynamicReloader.java [hunk 1]
```diff
@@ -242,9 +248,22 @@ private void reloadApp(AppReloadInfo appInfo) throws IOException {
         deployParam.set(DeploymentProperties.PATH, appInfo.getApplicationDirectory().getCanonicalPath());
         deployParam.set(DeploymentProperties.NAME, appInfo.getApplication().getName());
         deployParam.set(DeploymentProperties.KEEP_REPOSITORY_DIRECTORY, "true");
+
+        Properties reloadFile = appInfo.readReloadFile();
+        boolean hotDeploy = Boolean.TRUE.toString().equals(reloadFile.get(DeployCommandParameters.ParameterNames.HOT_DEPLOY));
```
Comment (OndroMih) [inline nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/DynamicReloader.java:253]: Why not use `Boolean.parseBoolean`?
```suggestion
        boolean hotDeploy = Boolean.parseBoolean(reloadFile.get(DeployCommandParameters.ParameterNames.HOT_DEPLOY));
```

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/DynamicReloader.java [hunk 2]
```diff
@@ -242,9 +248,22 @@ private void reloadApp(AppReloadInfo appInfo) throws IOException {
...
+        if (hotDeploy) {
+            deployParam.set(DeployCommandParameters.ParameterNames.HOT_DEPLOY, "true");
+            boolean metadataChanged = Boolean.TRUE.toString().equals(reloadFile.getProperty(DeployCommandParameters.ParameterNames.METADATA_CHANGED));
```
Comment (OndroMih) [inline nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/DynamicReloader.java:256]:
```suggestion
            boolean metadataChanged = Boolean.parseBoolean(reloadFile.getProperty(DeployCommandParameters.ParameterNames.METADATA_CHANGED));
```

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/DynamicReloader.java [hunk 3]
```diff
@@ -305,6 +324,16 @@ private boolean needsReload() {
             return answer;
         }
         
+        private Properties readReloadFile() {
+            Properties prop = new Properties();
+            try {
+                prop.load(new FileInputStream(reloadFile));
+            } catch (Exception ex) {
+                // skip
+            }
```
Comment (OndroMih) [inline nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/DynamicReloader.java:333]: The input stream should be closed. The [Javadoc](https://docs.oracle.com/javase/8/docs/api/java/util/Properties.html#load-java.io.InputStream-) states: "The specified stream remains open after this method returns."
```suggestion
            try (InputStream istream = new FileInputStream(reloadFile) {
                prop.load(istream);
            } catch (Exception ex) {
                // skip
            }
```

**Review by OndroMih — CHANGES_REQUESTED** [review]

**Review by MeroRai — APPROVED** [review]

**Comment by lprimak** [comment]
@OndroMih Looks like your review was satisfied no?

**Comment by lprimak** [comment]
jenkins test

**Comment by lprimak** [comment]
Build failed...

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/DynamicReloader.java [hunk 4]
```diff
@@ -305,6 +324,16 @@ private boolean needsReload() {
             return answer;
         }
         
+        private Properties readReloadFile() {
+            Properties prop = new Properties();
+            try (InputStream istream = new FileInputStream(reloadFile) {
```
Comment (lprimak) [inline nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/DynamicReloader.java:329]: missing paren here

**Comment by jGauravGupta** [comment]
jenkins test please

**Comment by jGauravGupta** [comment]
jenkins test please

**Review by OndroMih — APPROVED** [review]

**Pull request merged**